File: JavaScript/010.js
// Problem 10
//
// The sum of the primes below 10 is 2 + 3 + 5 + 7 = 17.
//
// Find the sum of all the primes below two million.

var limit = 2000000;
// var limit = 10;
var prime = 3;
var sum = 2;

while (prime < limit) {
  timesMultiplied = 0;
  for(var j = Math.round(Math.sqrt(prime)); j >= 1; j--) {
    if( prime % j == 0) {
      timesMultiplied++
    }
  }
  if(timesMultiplied < 2) {
    sum += prime
  }
  prime += 2
}

console.log(sum)

File: Python/010.py
# Problem 10
#
# The sum of the primes below 10 is 2 + 3 + 5 + 7 = 17.
#
# Find the sum of all the primes below two million.

import math

n = 3
sum = 2

def isPrime(n):
    if n % math.sqrt(n) == 0:
        return False
    limit = int(math.sqrt(n)) + 1
    for i in range(2, limit):
        if n % i == 0:
            return False

    return True


while n < 2000000:
    if isPrime(n):
        sum += n
    n += 2
print(sum)
